packages/api/internal/orchestrator/node.go

```diff
@@ -104,10 +107,10 @@ func (n *Node) setStatus(status api.NodeStatus) {
 	}
 }
 
-func (n *Node) setMetadata(i *orchestratorinfo.ServiceInfoResponse, nodeID string) {
+func (n *Node) setMetadata(i *orchestratorinfo.ServiceInfoResponse, nodeID string, serviceInstanceID string) {
```

Comment:
do you need to pass serviceInstanceID as extra argument instead of taking it from `ServiceInfoResponse`?